### ISSUE TITLE
Simplifying the learning curve for MMSeg

### ISSUE THREAD

**Issue by joshmyersdean** [open-mmlab/mmsegmentation#2553]
Hello!

First, thank you for all the work into this library. 
I am writing as I have been having trouble getting up and running with MMSeg, or at least getting to a point where I can prototype as quickly as in vanilla PyTorch.

Is there a way to initialize models without a monolithic config structure? I find it hard to have granular control of my model training and building with the config structure, especially for prototyping. For example, if I wanted to apply a loss to inner features of a network, it is not immediately clear on how to do so. In addition, is there any way to have the same control over training as a vanilla PyTorch training loop? Finally, when constructing datasets, how can I return multiple objects like one would in a `__getitem__` function? This is not immediately clear in the tutorials.

I'm sure I will eventually take advantage of the rich config structure, but is it possible to subvert it for quick prototyping?  

Thank you for your time!
